packages/valory/skills/market_creation_manager_abci/behaviours.py

```diff
@@ -1388,6 +1390,110 @@ def async_act(self) -> Generator:
         self.set_done()
 
 
+class PostTransactionBehaviour(MarketCreationManagerBaseBehaviour):
+    """A behaviour that is called after a transaction has been settled."""
+
+    matching_round: Type[AbstractRound] = PostTransactionRound
+
+    def async_act(self) -> Generator:
+        """Do the act, supporting asynchronous execution."""
+        with self.context.benchmark_tool.measure(self.behaviour_id).local():
+            sender = self.context.agent_address
+            content = yield from self.get_payload()
+            payload = PostTxPayload(sender=sender, content=content)
+        with self.context.benchmark_tool.measure(self.behaviour_id).consensus():
+            yield from self.send_a2a_transaction(payload)
+            yield from self.wait_until_round_end()
+        self.set_done()
+
+    def get_payload(self) -> Generator[None, None, str]:
+        """Get the transaction payload"""
+        settled_tx_hash = self.synchronized_data.settled_tx_hash
+        if settled_tx_hash is None:
+            self.context.logger.info("No settled tx hash.")
+            return PostTransactionRound.DONE_PAYLOAD
+
+        data = self.synchronized_data.approved_question_data
+        market_id = data.get("id", None)
+        if market_id is None:
+            self.context.logger.info("No market id.")
+            return PostTransactionRound.DONE_PAYLOAD
+
+        self.context.logger.info(f"Handling settled tx hash {settled_tx_hash}.")
+
+        if (
+            self.synchronized_data.tx_sender
+            != PrepareTransactionBehaviour.matching_round.round_id
+        ):
+            # we only handle market creation txs atm, any other tx, we don't need to take action
+            return PostTransactionRound.DONE_PAYLOAD
```

Comment:
`DONE_PAYLOAD` is actually not necessary as seen in the round's implementation. We could simply return `True` or `False` for done or error. The round would be a voting round, and by using the corresponding attributes we would not need to override the `end_block` method. Making use of these features of the framework would simplify both the behaviour and the round implementations.

Reply:
@Adamantios my idea was that this could have different payloads in the future depending on which round submitted the transaction.